sotodlib/site_pipeline/update_obsdbcols.py

```diff
@@ -0,0 +1,129 @@
+"""update_obsdbcols.py
+
+Add columns to entries of an obsdb for books that are also in the corresponding obsfiledb.
+The config file could be of the form:
+
+.. code-block:: yaml
+    obsdb_extra_cols: List or single string with form {name}: {type}
+    obsdb: dummyobsdb.sqlite
+    obsfiledb: dummyobsfiledb.sqlite
+"""
+
+from sotodlib.core.metadata import ObsDb, ObsFileDb
+from sotodlib.core import Context 
+from sotodlib.site_pipeline.check_book import main as checkbook
+from sotodlib.io import load_book
+import os
+import glob
+import yaml
+import numpy as np
+import time
+import argparse
+import logging
+from sotodlib.site_pipeline import util
+from typing import Optional
+
+logger = util.init_logger(__name__, 'update-obsdb: ')
+
+def main(config: str, 
+        recency: float = None, 
+        verbosity: Optional[int] = 2,
+        logger=None):
+
+    """
+    Create or update an obsdb for observation or operations data.
+
+    Arguments
+    ----------
+    config : str
+        Path to config file
+    recency : float
+        How far back in time to look for databases, in days. If None, 
+        goes back to the UNIX start date (default: None)
+    verbosity : int
+        Output verbosity. 0:Error, 1:Warning, 2:Info(default), 3:Debug
+    logger : logging
+        When to output the print statements
+
+    """
+
+    if logger is None:
+        logger = globals()['logger']
+    else:
+        globals()['logger'] = logger
+    if verbosity == 0:
+        logger.setLevel(logging.ERROR)
+    elif verbosity == 1:
+        logger.setLevel(logging.WARNING)
+    elif verbosity == 2:
+        logger.setLevel(logging.INFO)
+    elif verbosity == 3:
+        logger.setLevel(logging.DEBUG)
+
+    logger.info("Updating existing obsdb with new columns")
+
+    config_dict = yaml.safe_load(open(config, "r"))
+
+    if "obsdb" in config_dict:
+        obsdb = ObsDb(map_file=config_dict["obsdb"])
+    else:
+        logger.error("No obsdb named in the configuration file")
+        return
+
+    if "obsfiledb" in config_dict:
+        obsfiledb = ObsFileDb(map_file=config_dict["obsfiledb"])
+    else:
+        logger.error("No obsfiledb specified in the configuration file")
+        return
+    
+    if "obsdb_extra_cols" in config_dict:
+        col_list = []
+        for col, typ in config_dict["obsdb_extra_cols"].items():
+            col_list.append(col+" "+typ)
+        obsdb.add_obs_columns(col_list)
+    else: 
+        logger.error("No obsdb_extra_cols specified in the configuration file")
+        return
+
+    #How far back we should look
+    tnow = time.time()
+    if recency is not None:
+        tback = tnow - recency*86400
+    else:
+        tback = 0 #Back to the UNIX Big Bang 
+    
+    obs_ids = obsfiledb.get_obs()
+    for obs_id in obs_ids:
+        if obsdb.get(obs_id) is None:
+            logger.info(f"{obs_id} in obsfiledb but not in obsdb, skipping")
```

Comment:
there might be a point where we want to add new entries to the `obsdb` if we find them in the `obsfiledb`. is it possible to have this work here?